api/v1alpha1/gateway_types.go

```diff
@@ -270,28 +270,22 @@ type GatewayStatus struct {
 type GatewayConditionType string
 
 const (
-	// ConditionNoSuchGatewayClass indicates that the specified GatewayClass
-	// does not exist.
-	ConditionNoSuchGatewayClass GatewayConditionType = "NoSuchGatewayClass"
-	// ConditionGatewayNotScheduled indicates that the Gateway has not been
-	// scheduled.
-	ConditionGatewayNotScheduled GatewayConditionType = "GatewayNotScheduled"
-	// ConditionListenersNotReady indicates that at least one of the specified
-	// listeners is not ready. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionListenersNotReady GatewayConditionType = "ListenersNotReady"
-	// ConditionInvalidListeners indicates that at least one of the specified
-	// listeners is invalid. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionInvalidListeners GatewayConditionType = "InvalidListeners"
-	// ConditionRoutesNotReady indicates that at least one of the specified
-	// routes is not ready.
-	ConditionRoutesNotReady GatewayConditionType = "RoutesNotReady"
-	// ConditionInvalidRoutes indicates that at least one of the specified
-	// routes is invalid.
-	ConditionInvalidRoutes GatewayConditionType = "InvalidRoutes"
+	// ConditionGatewayClassExists indicates that the specified GatewayClass exists.
+	ConditionGatewayClassExists GatewayConditionType = "GatewayClassExists"
+	// ConditionGatewayScheduled indicates that the Gateway has been scheduled.
+	ConditionGatewayScheduled GatewayConditionType = "GatewayScheduled"
+	// ConditionListenersReady indicates that all of the specified listeners
+	// are ready. If this condition has a status of False, a more detailed
+	// ListenerCondition should be present in the corresponding ListenerStatus.
+	ConditionListenersReady GatewayConditionType = "ListenersReady"
+	// ConditionListenersValid indicates that all of the specified listeners
+	// are valid. If this condition has a status of False, a more detailed
+	// ListenerCondition should be present in the corresponding ListenerStatus.
+	ConditionListenersValid GatewayConditionType = "ListenersValid"
+	// ConditionRoutesReady indicates that all of the specified routes are ready.
```

Comment:
Can we combine routes ready with routes valid? (can you give a good example why we want to distinguish these two states)

Reply:
@bowei this comes from https://kubernetes.slack.com/archives/CR0H13KGA/p1584484692070800. A valid route means the config has passed validation. A ready route means some type of readiness probe has passed. I'm not opposed to sticking with valid listener/route now to simplify the conditions until a concrete use case is presented for adding a route ready state status condition. cc: @robscott

Reply:
My comments here are mostly around making the comments more descriptive. (e.g. can you take your responses and put them in the comments, so when people read it, they know exactly what the conditions mean?)